src/main/java/fi/aalto/cs/intellij/activities/RequiredPluginsCheckerActivity.java

```diff
@@ -0,0 +1,212 @@
+      if (!PluginManager.isPluginInstalled(descriptor.getPluginId())) {
+        missingPluginDescriptors.add(descriptor);
+      } else if (PluginManager.isDisabled(descriptor.getPluginId().getIdString())) {
+        disabledPluginDescriptors.add(descriptor);
+      }
+    }
+
+    if (missingPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsInstallation(missingPluginDescriptors);
+    } else if (disabledPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsEnabling(disabledPluginDescriptors);
+    }
+  }
+
+  /**
+   * Notify with with an option to enable all the required plugins.
+   */
+  private void notifyAndSuggestPluginsEnabling(List<IdeaPluginDescriptor> descriptors) {
+    Notification notification = new Notification(
+        "A+",
+        "A+",
+        "Some plugins must be and enabled for the A+ plugin to work properly "
+            + getPluginsNamesString(descriptors) + ".",
+        NotificationType.WARNING);
+
+    notification.addAction(new NotificationAction(
+        "Enable the required plugin(s) (" + getPluginsNamesString(descriptors) + ").") {
+
+      /**
+       * Activate all the required plugins.
+       */
+      @Override
+      public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
+        descriptors.forEach(descriptor -> Objects
+            .requireNonNull(PluginManager.getPlugin(descriptor.getPluginId())).setEnabled(true));
+        notification.expire();
+      }
+    });
+
+    Notifications.Bus.notify(notification);
+  }
+
+  /**
+   * Notify with with an option to install all the required plugins and suggest restart.
+   */
+  private void notifyAndSuggestPluginsInstallation(List<IdeaPluginDescriptor> descriptors) {
+    Notification notification = new Notification(
+        "A+",
+        "A+",
+        "Additional plugin(s) must be installed and enabled for the A+ plugin to work "
+            + "properly (" + getPluginsNamesString(descriptors) + ").",
+        NotificationType.WARNING);
+
+    notification.addAction(new NotificationAction(
+        "Install missing (" + getPluginsNamesString(descriptors) + ") plugin(s).") {
+
+      /**
+       * Install the missing plugins and propose a restart.
+       */
+      @Override
+      public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
+        descriptors.forEach(descriptor -> {
+          try {
+            PluginDownloader pluginDownloader = PluginDownloader.createDownloader(descriptor);
+            pluginDownloader.prepareToInstall(new BgProgressIndicator());
+            pluginDownloader.install();
+          } catch (IOException ex) {
+            logger.error("Could not install plugin" + descriptor.getName() + ".", ex);
+          }
+        });
+        notification.expire();
+        PluginManagerConfigurable
+            .shutdownOrRestartApp("Plugins required for A+ course are now installed");
```

Comment:
This is probably correct but I just don't get what is actually shutdown or restarted here. IDEA?

Reply:
it restarts the IDE ultimately. 

`PluginManagerConfigurable.shutdownOrRestartApp(...` is just a wrapper in PluginManagerConfigurable class that actually refers to a deeper service, that does the restart (as some parts of Plugin Management require restarting the IDE). 

for details see:
https://github.com/JetBrains/intellij-community/blob/f91f570d6ebf62c71742e99e7ac08a07ad4d17b3/platform/platform-impl/src/com/intellij/ide/plugins/PluginManagerConfigurable.java#L1315-L1323

Reply:
Oh right. And that text is the title of the popup message window, nice. I didn't notice it when I first tested.
![image](https://user-images.githubusercontent.com/6066364/72511359-b64a8700-3853-11ea-8a70-c3b6042c3cb6.png)